templates/common/gcp/units/gcp-update-dns.service.yaml

```diff
@@ -0,0 +1,19 @@
+name: gcp-update-dns.service
+enabled: {{if and (.Infra.Status.PlatformStatus.GCP) (.Infra.Status.PlatformStatus.GCP.CloudLoadBalancerConfig) (eq (.Infra.Status.PlatformStatus.GCP.CloudLoadBalancerConfig.DNSType .) "ClusterHosted") }}true{{else}}false{{end}}
+contents: |
+  [Unit]
+  Description=Update Default GCP Resolver
+  # We don't need to do this on the firstboot
+  After=firstboot-osupdate.target
+  # Wait for NetworkManager to report it's online
+  After=NetworkManager-wait-online.service
+  # TODO: Is this correct?
+  Before=network-online.target
```

Comment:
I would probably change this to kubelet-dependencies. We moved away from using network-online because it had some nasty transitive dependencies that caused a lot of grief.

Reply:
Thanks for that pointer.